docs/bundler/hmr.md

```diff
@@ -0,0 +1,266 @@
+| --- | ------------------ | --------------------------------------------------------------------------------- |
+| 🚧  | `hot.accept()`     | API exists but only logs "TODO" warning; accepts no module boundaries             |
+| ✅  | `hot.dispose()`    | Fully functional; registers callbacks that are executed before module replacement |
+| ✅  | `hot.data`         | Fully functional; persists data between module evaluations                        |
+| ❌  | `hot.decline()`    | Exists as empty function for compatibility with Vite APIs                         |
+| 🚧  | `hot.invalidate()` | Exists but throws "TODO: implement ImportMetaHot.invalidate" error                |
+| 🚧  | `hot.on()`         | Exists but throws "TODO: implement ImportMetaHot.on" error                        |
+| 🚧  | `hot.off()`        | Exists but throws "TODO: implement ImportMetaHot.off" error                       |
+| 🚧  | `hot.send()`       | Exists but throws "TODO: implement ImportMetaHot.send" error                      |
+| 🚧  | `hot.prune()`      | Exists but throws "TODO: implement ImportMetaHot.prune" error                     |
+
+**Currently Functional APIs:**
+
+- `hot.dispose()`: Successfully registers cleanup callbacks that run before module replacement
+- `hot.data`: Successfully maintains state between module evaluations
+
+### `hot.accept()`
+
+{% callout %}
+❌ **Not implemented** — Currently, `hot.accept()` logs a warning message and does not execute the provided callbacks. The module tree will still be recursively re-evaluated, but without respecting the specified acceptance boundaries.
+{% /callout %}
+
+Accepts hot updates for this module or its dependencies.
+
+```ts#hmr-accept.ts
+// Accept self (simplest form)
+import.meta.hot.accept();
+// This logs: "TODO: implement ImportMetaHot.accept" and does nothing
+
+// Accept self with a callback to handle updates
+import.meta.hot.accept(newModule => {
+  // This callback will not be called in the current implementation
+  console.log("Updated module received", newModule);
+});
+
+// Accept updates to specific dependencies
+import.meta.hot.accept(["./dep1.js", "./dep2.js"], ([dep1, dep2]) => {
+  // This callback will not be called in the current implementation
+  console.log("Dependencies updated");
+});
+```
+
+In future releases, when a module is accepted, Bun will properly respect the acceptance boundaries, preventing propagation of updates beyond accepted modules and executing the provided callbacks. This will enable more granular control over how module updates are handled.
+
+### `hot.dispose()`
+
+{% callout type="success" %}
+✅ **Fully implemented** — `hot.dispose()` works as expected, allowing you to register callbacks that will be executed before the module is replaced.
+{% /callout %}
+
+Registers a callback that will be called when the module is about to be replaced.
+
+```ts#dispose.ts
```

Comment:
this example is too abstract to me. a good example would include the initialization code